### PR TITLE
Refactor code in main app

### PR DESCRIPTION
**Pull request by cherryshek-1014**

### CODE REVIEW

File: pages/dictionary/search_history.tsx [hunk 1]
```diff
@@ -3,10 +3,22 @@ import styles from '>>/styles/dictionary.module.css'
...
+    searchList: String[]
 }
 
-const History = ({ searchWord }: Props) => {
+const History = ({ setDisplayText, searchList }: Props) => {
+    const jumpTo = (word) => {
+        setDisplayText(word);
+    }
+
+    const searchWord = searchList.map((searchText, index) => {
```
Comment (ColinCee) [inline pages/dictionary/search_history.tsx]: Rename to searchedWords
```suggestion
    const searchedWords = searchList.map((searchText, index) => {
```

File: pages/dictionary/search_history.tsx [hunk 2]
```diff
@@ -3,10 +3,22 @@ import styles from '>>/styles/dictionary.module.css'
...
-    searchWord: Map<string, number>
+    setDisplayText: React.Dispatch<React.SetStateAction<string>>
+    searchList: String[]
 }
 
-const History = ({ searchWord }: Props) => {
+const History = ({ setDisplayText, searchList }: Props) => {
+    const jumpTo = (word) => {
+        setDisplayText(word);
+    }
```
Comment (ColinCee) [inline pages/dictionary/search_history.tsx]: This seems redundant, you can just call setDisplayText in the below function

File: pages/dictionary/search_history.tsx [hunk 3]
```diff
@@ -3,10 +3,22 @@ import styles from '>>/styles/dictionary.module.css'
...
+
+    const searchWord = searchList.map((searchText, index) => {
+        return (
+            <li key={index}>
+                <button onClick={() => jumpTo(searchText)}>{searchText}</button>
```
Comment (ColinCee) [inline pages/dictionary/search_history.tsx]:
```suggestion
                <button onClick={() => setDisplayText(searchText)}>{searchText}</button>
```

**Review by ColinCee — CHANGES_REQUESTED** [review]

File: pages/dictionary/search_history.tsx (hunk 1 above)
Comment (cherryshek-1014) [reply pages/dictionary/search_history.tsx]: Fixed in [[fd7af37](https://github.com/cherryshek-1014/react-projects/pull/1/commits/fd7af37fc7ed7b98e1e3bc3d55b96ac146b789b8)](url)

File: pages/dictionary/search_history.tsx (hunk 2 above)
Comment (cherryshek-1014) [reply pages/dictionary/search_history.tsx]: Fixed in [[fd7af37](https://github.com/cherryshek-1014/react-projects/pull/1/commits/fd7af37fc7ed7b98e1e3bc3d55b96ac146b789b8)](url)

File: pages/dictionary/search_history.tsx (hunk 3 above)
Comment (cherryshek-1014) [reply pages/dictionary/search_history.tsx]: fixed in [[fd7af37](https://github.com/cherryshek-1014/react-projects/pull/1/commits/fd7af37fc7ed7b98e1e3bc3d55b96ac146b789b8)](url)

File: pages/dictionary/index.tsx [hunk 4]
```diff
@@ -10,37 +10,11 @@ const Dictionary = () => {
     const [displayText, setDisplayText] = useState("");
     const [searchList, setSearchList] = useState<string[]>([]);
```
Comment (ColinCee) [inline pages/dictionary/index.tsx:11]: Change this to searchedWords

File: pages/dictionary/search_history.tsx [hunk 5]
```diff
@@ -3,15 +3,24 @@ import styles from '>>/styles/dictionary.module.css'
...
 type Props = {
-    searchWord: Map<string, number>
+    setDisplayText: React.Dispatch<React.SetStateAction<string>>
+    searchList: String[]
 }
 
-const History = ({ searchWord }: Props) => {
+const History = ({ setDisplayText, searchList }: Props) => {
+
+    const searchedWords = searchList.map((searchText, index) => {
```
Comment (ColinCee) [inline pages/dictionary/search_history.tsx:12]: Rename this to searchHistory

File: pages/dictionary/index.tsx [hunk 6]
```diff
@@ -10,37 +10,11 @@ const Dictionary = () => {
...
-
-    const handleOnChange = (event) => {
-        setInputText(event.target.value);
-    }
-
-
     return (
         <div className={styles.container}>
-            <History searchWord={searchWord} />
+            <History setDisplayText={setDisplayText} searchList={searchList} />
```
Comment (ColinCee) [inline pages/dictionary/index.tsx]: Remember to update references to searchedWords

**Review by ColinCee — APPROVED** [review]

**Pull request merged**